sources/editor/Stride.Core.Assets.Editor/View/TemplateProviders/DictionaryNumberKeyTemplateProvider.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net) and Silicon Studio Corp. (https://www.siliconstudio.co.jp)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+using System;
+using System.Collections.Generic;
+using Stride.Core.Assets.Editor.Quantum.NodePresenters.Updaters;
+using Stride.Core.Presentation.Quantum;
+using Stride.Core.Presentation.Quantum.ViewModels;
+
+namespace Stride.Core.Assets.Editor.View.TemplateProviders
+{
+    public class DictionaryNumberKeyTemplateProvider : DictionaryTemplateProvider
```

Comment:
Can this be made more generic, to perhaps support any other value type ?

Reply:
This template provider here is just to support number value key dictionary to create new key shows default value as 0, I just reference the DictionaryEnmuKeyTemplateProvider to write this.
As my purpose is to fix the dictionary editor with value types key, so I decide to keep the old style of the code. So a better way with much modification is not in my consideration.
And also, other value type key can use default dictionary template with the default value, such as '' for char, false for bool and so on, I think it's not nessary to support so generic for the default has done this work.